### PR TITLE
gh-75729: Fix os.spawn tests not handling spaces on Windows

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
As originally reported in #75729 and confirmed to still be reproducible on the latest Python 3.12, the various `os.spawn*` tests are broken on Windows if the current working directory contains any spaces in its path (e.g. due to spaces in the username), as on Windows the arguments are simply concatenated and not quoted. This PR fixes the immediate issue with the tests by quoting the arguments in question on Windows.

* Issue: gh-75729

### CODE REVIEW

**Comment by CAM-Gerlach** [comment]
Should this have a NEWS entry? I see some PRs that only change the tests have them, but many (most) I spot-checked don't.

**Comment by JelleZijlstra** [comment]
I think a NEWS entry is helpful here because this could affect redistributors or users who are trying to run the tests locally.

**Comment by CAM-Gerlach** [comment]
Thanks, I added one, but it seems there's something funny going on with GitHub, as when I pushed it failed with a strange error, even after I reset and re-committed:

```
remote: fatal error in commit_refs
To github.com:CAM-Gerlach/cpython.git
 ! [remote rejected]         fix-spawn-tests-space-in-path -> fix-spawn-tests-space-in-path (failure)
error: failed to push some refs to 'github.com:CAM-Gerlach/cpython.git'
```

I then force pushed and my fork updated, but nothing happened here for several minutes. Finally I rebased and force-pushed again, and it appears to finally have worked.

File: Lib/test/test_os.py [hunk 1]
```diff
@@ -3089,6 +3089,14 @@ def kill_process(pid):
 
 @support.requires_subprocess()
 class SpawnTests(unittest.TestCase):
+    @staticmethod
+    def quote_args(args):
+        # On Windows, os.spawn* simply joins arguments with spaces:
+        # arguments need to be quoted
+        if not os.name == 'nt':
```
Comment (JelleZijlstra) [inline Lib/test/test_os.py]:
```suggestion
        if os.name != 'nt':
```

File: Lib/test/test_os.py [hunk 2]
```diff
@@ -3089,6 +3089,14 @@ def kill_process(pid):
...
+            return args
+        return [f'"{arg}"' if " " in arg.strip() else arg for arg in args]
```
Comment (JelleZijlstra) [inline Lib/test/test_os.py:3223]: What if there is a double quote in the path?

File: Lib/test/test_os.py (hunk 1 above)
Comment (CAM-Gerlach) [reply Lib/test/test_os.py]: Of course, silly me

File: Lib/test/test_os.py (hunk 2 above)
Comment (CAM-Gerlach) [reply Lib/test/test_os.py:3223]: That's not possible on Windows, as the double quote character [is not allowed in Windows file paths](https://en.wikipedia.org/wiki/Filename#In_Windows), so this quoting strategy is sufficient for its purpose here. If one did need to handle arbitrary arguments, the undocumented [`subprocess.list2cmdline`](https://github.com/python/cpython/blob/bef9efabc3f899a8f05cc6bee009c7c5fb3d01ae/Lib/subprocess.py#L576) could be used instead.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Does os.exec tests need a similar fix?

**Comment by CAM-Gerlach** [comment]
Good question, and at least as far as the scope of this PR, the `os.spawn*` tests are the only ones affected in the `os` module (there are a couple unrelated tests affected in other modules, which I will be opening separate issues/PRs for), when tested with both the interpreter path, the working dir path and the temp dir path all containing spaces (given they are all under a user directory containing such).

However, the `os.exec*` functions on Windows _do_ have the same issue as the `os.spawn*` functions, a fact I will note on the corresponding issue—but their tests don't, because as far as I could find, there apparently _are_ no tests of `os.exec*` where this issue would be triggered, since the only tests are for empty, invalid, etc. arguments that return with an exception before the bad behavior would be triggered at the OS level. I'm guessing this is because if `os.exec*` actually executed successfully, this would naturally break the tests, since the testing process would get replaced with whatever is launched.

**Comment by hugovk** [comment]
Is this ready to merge?

**Comment by CAM-Gerlach** [comment]
It should be as far as I'm aware, yeah—there are some other issues that need another round of discussion, but we can at least fix the tests for now.

**Review by hugovk — APPROVED** [review]

**Pull request merged**